src/java/org/apache/cassandra/net/RequestCallbacks.java

```diff
@@ -338,7 +342,7 @@ static class WriteCallbackInfo extends CallbackInfo
 
         public boolean shouldHint()
         {
-            return mutation != null && StorageProxy.shouldHint(replica);
+            return shouldHint && StorageProxy.shouldHint(replica);
```

Comment:
This (in addition to `getIMutation on line 362) is the "safest workaround" I could make to carry around a non-nullable Mutation object. 

I dunno the history here but I think mutation nullability was dependent on `shouldHint` is to be memory savvy. For Sensors we need this object at all times. We could add another object or just modify the behavior here to be `shouldHint` independent. I opted for the latter because judging by tests that failed (on C* & CNDB) majority of use cases use `shouldHint` so instead of introducing another mutation object for sensors (Which would be redundant most of the time) I thought to decouple mutation nullability from `shouldHint`